### PR TITLE
Whitespace, comments etc. tidy up before migration to JUnit 5

### PR DESCRIPTION
**Pull request by ham1**
Formatting/tidy up PR, before committing for review JUnit 5 work.

Sorry it's so long but it the JUnit 5 changes touch a lot of files and don't want these changes causing and confusion or delay.

### CODE REVIEW

**Comment by vlsi** [comment]
I wonder if "remove optional semicolon in Groovy" can be configured with Spotless.

File: src/components/src/test/java/org/apache/jmeter/gui/action/TestSave.java [hunk 1]
```diff
@@ -28,25 +28,24 @@
 import org.junit.Test;
 
 public class TestSave {
-        private Save save;
+    private Save save;
 
+    @BeforeEach
```
Comment (vlsi) [inline src/components/src/test/java/org/apache/jmeter/gui/action/TestSave.java]: Is this a bit of JUnit5?

File: src/components/src/test/java/org/apache/jmeter/timers/SyncTimerTest.java [hunk 2]
```diff
@@ -88,8 +88,7 @@ public void testTimerWithInvalidTimeout() {
     private long timeDelay(SyncTimer timer) {
         long start = System.currentTimeMillis();
         timer.delay();
-        long duration = System.currentTimeMillis() - start;
-        return duration;
+        return System.currentTimeMillis() - start;
```
Comment (vlsi) [inline src/components/src/test/java/org/apache/jmeter/timers/SyncTimerTest.java]: Technically speaking, those "one-time" variables might help to debug the values.
I can set a breakpoint "at the return statement" and see the delay value.
As the variable is inlined the same breakpoint might become harder.

I don't see how this change simplifies JUnit5->JUnit4 migration :-/

File: src/components/src/test/java/org/apache/jmeter/timers/poissonarrivals/PreciseThroughputTimerTest.java [hunk 3]
```diff
@@ -68,20 +69,10 @@ public void testExactNumberOfSamples() throws Exception {
     protected ConstantPoissonProcessGenerator getConstantPoissonProcessGenerator(
             final double throughput, final int duration, long seed) {
         return new ConstantPoissonProcessGenerator(
-                new ThroughputProvider() {
-                    @Override
-                    public double getThroughput() {
-                        return throughput; // samples per second
-                    }
-                },
+                () -> throughput,
```
Comment (vlsi) [inline src/components/src/test/java/org/apache/jmeter/timers/poissonarrivals/PreciseThroughputTimerTest.java]: Did you replace this manually?
Did you use an IDE action to automatically find and replace it?

File: src/components/src/test/groovy/org/apache/jmeter/extractor/json/render/RenderAsJsonRendererSpec.groovy [hunk 4]
```diff
@@ -112,7 +108,7 @@ class RenderAsJsonRendererSpec extends JMeterSpec {
     def "setupTabPane adds the tab to rightSide"() {
         given:
             sut.init()
-            def rightSideTabbedPane = new JTabbedPane();
+            def rightSideTabbedPane = new JTabbedPane()
```
Comment (vlsi) [inline src/components/src/test/groovy/org/apache/jmeter/extractor/json/render/RenderAsJsonRendererSpec.groovy]: I wish this "optional semicolon" was represented in Spotless somehow.

**Comment by vlsi** [comment]
@ham1 , do you really intend to touch `*.groovy` files in JUnit5?

Even though the changes you propose look good, and they improve things, it is tricky that current Spotless/Checkstyle are happy with the current code.
That means new contributions might easily add code with the same formatting that you have just removed. Are you going to continuously format incoming code?

For instance: "optional semicolons in Groovy". I see you removed certain items, however, you don't seem to add the relevant Spotless configuration.

PS. It looks like you've committed Before->BerforeEach which caused Travis failure.

File: src/components/src/test/java/org/apache/jmeter/gui/action/TestSave.java (hunk 1 above)
Comment (ham1) [reply src/components/src/test/java/org/apache/jmeter/gui/action/TestSave.java]: Yeah, will fix that, thank you for spotting it.

File: src/components/src/test/java/org/apache/jmeter/timers/SyncTimerTest.java (hunk 2 above)
Comment (ham1) [reply src/components/src/test/java/org/apache/jmeter/timers/SyncTimerTest.java]: Yeah you are correct, this one doesn't relate to JUnit5, was just a simplification of the code, as I don't think the name `duration` is especially helpful given the method is short and called `timeDelay`. Regarding debugging, you could also just step out of the function and again as it's a shot method this one didn't seem like it would need a lot of debugging.

Happy to revert this change it you'd prefer.

File: src/components/src/test/java/org/apache/jmeter/timers/poissonarrivals/PreciseThroughputTimerTest.java (hunk 3 above)
Comment (ham1) [reply src/components/src/test/java/org/apache/jmeter/timers/poissonarrivals/PreciseThroughputTimerTest.java]: IntelliJ suggested it, however it was a 2 step process, first it changed it to:

```
() -> {
    return throughput; // samples per second
}
```
Then I removed the comment and it correct it to `() -> throughput`.

File: src/components/src/test/groovy/org/apache/jmeter/extractor/json/render/RenderAsJsonRendererSpec.groovy (hunk 4 above)
Comment (ham1) [reply src/components/src/test/groovy/org/apache/jmeter/extractor/json/render/RenderAsJsonRendererSpec.groovy]: Me, too, maybe a suggestion for them?

**Comment by ham1** [comment]
I didn't intend to touch Spock tests, however I did accidentally when changing `@Ignore` to `@Disabled` and while fixing that issue I also tidied up some of the files.

You are correct, I'm not going to continually format the code, but I can't not do it when I see it :smile: and I think it's better to make changes for the better so that:
1. others see it and might also learn/follow and
2. it's better than giving up and making the entropy continually worse

I will have a look into the Groovy-Eclipse formatter for Spotless.

File: src/components/src/test/java/org/apache/jmeter/timers/SyncTimerTest.java (hunk 2 above)
Comment (vlsi) [reply src/components/src/test/java/org/apache/jmeter/timers/SyncTimerTest.java]: I see where are you heading, and those small changes are good.

However, they might interfere with other PRs. See https://github.com/apache/jmeter/pull/484

The title says "whitespace, comments tidy", and variable inlining is neither whitespace nor comments.

My opinion here would be:
1) It is ok to keep this change in this PR (it is unlikely someone would face a merge conflict over this particular line)
2) Please refrain from making unrelated changes in your further contributions. Here you have **single** commit that touches lots of lines. Suppose someone would be bitten by a merge conflict. Of course, they might try to diff what "you intended to change", and they might be overwhelmed by 5000 line diff of **different** styles of changes.

In other words, it is OK to have 5000 lines diff that implements "remove trailing semicolon" action (so even in case of merge conflicts it is clear what should be done), however, it is not OK to have 1000 lines diff with three unrelated changes in a single commit.


@pmouawad , comments, thoughts?

File: src/components/src/test/java/org/apache/jmeter/timers/poissonarrivals/PreciseThroughputTimerTest.java (hunk 3 above)
Comment (vlsi) [reply src/components/src/test/java/org/apache/jmeter/timers/poissonarrivals/PreciseThroughputTimerTest.java]: In IntelliJ IDEA:
1. Help -> "Find action..."
2. "Run inspection by name..."
3. "Anonymous type can be replaced with lambda"
4. "whole project", "include test sourecs"

It produces a list of 20 or so items to be replaced.
It would be so much better if you included that change in a separate commit.
All of those changes could go under `Apply "Anonymous type can be replaced with lambda" inspection` commit.

WDYT?

File: src/components/src/test/java/org/apache/jmeter/timers/SyncTimerTest.java (hunk 2 above)
Comment (ham1) [reply src/components/src/test/java/org/apache/jmeter/timers/SyncTimerTest.java]: Yes, this makes sense, thank you for taking the time to explain; I will try to keep changes specific to a single PR. I'll remove this change for now.

**Comment by vlsi** [comment]
LTGM

File: src/components/src/test/groovy/org/apache/jmeter/extractor/json/render/RenderAsJsonRendererSpec.groovy (hunk 4 above)
Comment (ham1) [reply src/components/src/test/groovy/org/apache/jmeter/extractor/json/render/RenderAsJsonRendererSpec.groovy]: See #518

**Comment by ham1** [comment]
I've just rebased, is someone ok to merge this so I can show the JUnit 5 migration? :smile:

**Comment by vlsi** [comment]
@ham1 , I see you contribute a lot. This is nice.
I have a lifehack for you: you can unlock Travis in your own JMeter fork https://travis-ci.org/ham1/jmeter
It might provide you with a faster turnaround.

The thing is all Apache projects have a limited set of Travis runners, and it often happens that it takes a lot to wait for Travis job to even start (because the runners are busy with building other ASF projects).

**Comment by ham1** [comment]
That is a good tip, thanks, I hadn't realised that was possible.
